### ISSUE TITLE
Direct import of json recipe

### ISSUE THREAD

**Issue by Ultchad** [nextcloud/cookbook#1671]
Would it be possible to directly import json files? (Example from your git)

**Failing website:** https://raw.githubusercontent.com/nextcloud/cookbook/1c60fc03cda4c4af3e650637d5f3b22101e37c6f/tests/Unit/Helper/HTMLParser/res_JsonLd/caseJ.json

**Checking if valid metadata are present:**
 [x]  Yes, I check the source code of the website and found metadata. it's directly the json shema

**Cookbook version:** 10.2

**Problem description (if applicable):**
The page cannot be imported. The error message is:
"No parser found for the given import."

Thanks for this project!

Actualy i create an empty recipe (from your app) and i edit the `recipe.json` (of the new recipe) with the json schema I found

**Comment by christianlupus** [comment]
Yes, in general it is possible. One has to wipe a new importer class. This should not be too much effort, i guess. I have to check, though.

May i ask why you try to import directly? Is there a page that offers the JSON files explicitly?

**Comment by Ultchad** [comment]
The recipe page loads the json file from a network call and I get the url of the json.
Other pages have 2, when loading the page a script get the json via the network, then integrates it into the page (the cookbook import does not currently work on it, i have the same error, i can create other issue later for this site)

**Comment by doguitar** [comment]
> May i ask why you try to import directly? Is there a page that offers the JSON files explicitly?

ChatGPT is great at reformatting recipes to schema.org format. I've been manually adding recipes to the nextcloud filesystem with it but it would be amazing if I could just paste in the json.

**Comment by magarnicle** [comment]
https://github.com/hhursev/recipe-scrapers can already scrape dozens of websites into schema.org format, so this feature would make it very easy to integrate with that. Exposing the import function somehow so that it is callable would be even better :)

**Comment by gewinh** [comment]
Would certainly be a great feature I have used Chatgpt to reformat an unstructured recipe into schema .org format to get output like (see below). To be able to import this directly, would be ideal.

<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Recipe",
  "name": "2 HOUR Crockpot Bread",
  "author": "Pamela Reed",
  "description": "Easy to make 2 HOUR Crockpot Bread recipe. Throw it in the crock pot and you will have soft homemade bread in 2 hours! This is the best bread to serve with creamy soups and casseroles.",
  "prepTime": "PT45M",
  "cookTime": "PT2H",
  "recipeYield": "10 servings",
  "recipeIngredient": [
    "1 cup warm water",
    "1/3 cup sugar",
    "1 tablespoon active dry yeast",
    "1 1/2 teaspoon salt",
    "2 tablespoons vegetable oil",
    "3 cups all-purpose flour",
    "rosemary (optional)"
  ],
  "recipeInstructions": [
    "Pour warm water into a large bowl. Add sugar and mix until dissolved. Add dry yeast and stir. Let sit for about 10 minutes until foamy. I always find it helps to sit it in a warm spot.",
    "Add salt to the bowl and stir to dissolve. Add vegetable oil.",
    "Add flour and mix with hands until everything is combined. Knead the dough for 5 minutes and form into a round ball.",
    "Place the dough in an olive oil-coated bowl and cover with plastic wrap. Sit in a warm spot for 30 minutes to rise, doubling in size.",
    "Cut a piece of parchment paper and place it in the bottom of the slow cooker. Make sure the parchment paper comes up the bottom of the sides of the slow cooker so when the bread expands, it won't stick to the sides. Additionally, I like to spray the parchment paper with nonstick spray.",
    "Place your dough into the slow cooker. Sprinkle with rosemary.",
    "Cook on HIGH for 2 hours.",
    "Remove from the slow cooker and brush a little bit of butter on top. Place under the broiler for about 1 minute until slightly brown. Keep an eye on it and make sure it does not burn! Broiling is not a necessary step, but I recommend it to get the top crispy!"
  ]
}
</script>

**Comment by christianlupus** [comment]
Okay, to be honest I am a bit unsure if I understand you correctly of if this is in fact two feature requests. 

@Ultchad, @gewinh, and all the others involved: what exactly are you looking for, a dringend it a backend solution? That is, do you want to have an UI text field where you can copy and paste e.g. the output of ChatGPT? Or are you missing an option to allow the parser to detect plain JSON+LD files directly?

My original interpretation was the latter (assignment the parser capabilities). Please clarify, we might open another issue if needed.

**Comment by magarnicle** [comment]
My request was more the latter - I was trying to migrate from a different service and I had to create HTML files that wrapped the JSON, stand up a web server to serve them so that Cookbook could import them. Pointing Cookbook and the JSON files would have skipped a bunch of steps. An API or command-line call to import a batch of recipes would be even better but more work, obviously.

**Comment by christianlupus** [comment]
@magarnicle the current API should be capable of creating new recipes by uploading the JSON data plainly. I guess there is nothing fancy needed for this requirement. 

@doguitar I would keep your request separately as it is related to UI while the other two parts are more backend oriented.

**Comment by gewinh** [comment]
Thanks for the follow up. Either import a JSON file or cut and paste it to the ui would be great.

**Comment by Ultchad** [comment]
> Okay, to be honest I am a bit unsure if I understand you correctly of if this is in fact two feature requests.
> 
> @Ultchad, @gewinh, and all the others involved: what exactly are you looking for, a dringend it a backend solution? That is, do you want to have an UI text field where you can copy and paste e.g. the output of ChatGPT? Or are you missing an option to allow the parser to detect plain JSON+LD files directly?
> 
> My original interpretation was the latter (assignment the parser capabilities). Please clarify, we might open another issue if needed.

Actualy it's for import directly (with the field to download a recipe from an URL in webui) a JSON file, by giving the url [like my example]( https://raw.githubusercontent.com/nextcloud/cookbook/1c60fc03cda4c4af3e650637d5f3b22101e37c6f/tests/Unit/Helper/HTMLParser/res_JsonLd/caseJ.json)

Sorry for my late response
